aten/src/ATen/core/TensorBody.h

```diff
@@ -781,10 +781,10 @@ class CAFFE2_API Tensor {
   int64_t q_per_channel_axis() const;
   Tensor int_repr() const;
   QScheme qscheme() const;
-  Tensor to(const TensorOptions & options, bool non_blocking=false, bool copy=false) const;
-  Tensor to(Device device, ScalarType dtype, bool non_blocking=false, bool copy=false) const;
-  Tensor to(ScalarType dtype, bool non_blocking=false, bool copy=false) const;
-  Tensor to(const Tensor & other, bool non_blocking=false, bool copy=false) const;
+  Tensor to(const TensorOptions & options, bool non_blocking=false, bool copy=false, c10::optional<MemoryFormat> memory_format=c10::nullopt) const;
```

Comment:
I told you this before, and I'm going to complain about it again here: MemoryFormat should be a parameter in TensorOptions, not a freestanding positional argument as it is here. The state of positional arguments here is basically unusable for actual users of this API.

Reply:
Going to be merged into TensorOptions as soon as TensorOptions codegen rewrite finished ( CC @izdeby )